corehq/sql_accessors/sql_templates/get_case_ids_modified_with_owner_since.sql

```diff
@@ -0,0 +1,19 @@
+DROP FUNCTION IF EXISTS get_case_ids_modified_with_owner_since(text, text, timestamp with time zone);
+
+CREATE FUNCTION get_case_ids_modified_with_owner_since(domain_name text, p_owner_id text, reference_date timestamp with time zone)
+RETURNS TABLE (case_id VARCHAR(255)) AS $$
+BEGIN
+    RETURN QUERY
+    SELECT case_table.case_id FROM form_processor_commcarecasesql as case_table
+    WHERE case_table.domain = domain_name
+      AND case_table.server_modified_on >= reference_date
+      -- AND owner_id matches OR (owner_id is falsey AND modified_by matches)
+      AND (
+         case_table.owner_id = p_owner_id
+         OR (
```

Comment:
This is probably ok for now, but I think we definitely want to address updating this in another PR.  I think this `OR` here with reference to `modified_by` will basically make us lose the advantage of the index on `owner_id`, so if we just set `owner_id` appropriately on the initial case migration we can join exclusively on `owner_id` and that would be a lot faster.